create-publication-from-scopus/src/test/java/no/sikt/nva/scopus/test/utils/ScopusGenerator.java

```diff
@@ -58,13 +60,14 @@
     private static final Set<String> IGNORED_FIELDS = readIgnoredFields();
     private final DocTp document;
 
-    public ScopusGenerator() {
-        this.document = randomDocument();
+
+    public ScopusGenerator(CitationtypeAtt citationtypeAtt) {
+        this.document = randomDocument(citationtypeAtt);
     }
 
-    public static DocTp randomDocument() {
+    public static DocTp randomDocument(CitationtypeAtt citationtypeAtt) {
```

Comment:
This will not allow us to create a randomDocument when citationType is not important. Keep the public constructor empty, and create 
1. a field of type CitationTypeAtt
2. a private contructor that accepts that field
3. a public static method to create a ScopusGenerator with the selected CitationTypeAtt
4. modify the generation of the citationType to be either the non null field or a random item.

Reply:
fixed